libp2p/protocols/secure/secio.nim

```diff
@@ -404,8 +404,8 @@ proc readLoop(sconn: SecureConnection, stream: BufferStream) {.async.} =
     try:
       let msg = await sconn.readMessage()
       await stream.pushTo(msg)
-    except CatchableError as exc:
-      trace "exception in secio", exc = exc
+    except Exception as exc:
```

Comment:
While the other cases of handling and re-raising `Exceptions` seem OK to me, recovering from `Defects` here doesn't seem appropriate. I've pushed another commit changing this back to `CatchableError`.